src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -1039,8 +1086,9 @@ LIMIT 1
             commandText: @"
 SELECT nfid.non_fungible_id AS NonFungibleId, final.total_count AS NonFungibleIdsTotalCount
 FROM (
-    SELECT UNNEST(non_fungible_ids[@offset:@limit]) AS non_fungible_id_data_id, cardinality(non_fungible_ids) AS total_count
+    SELECT non_fungible_id_data_id, cardinality(non_fungible_ids) AS total_count, ordinality
     FROM entity_vault_history
+    LEFT JOIN LATERAL UNNEST(non_fungible_ids[@offset:@limit]) WITH ORDINALITY as non_fungible_id_data_id on true
```

Comment:
Open question: wouldn't it be more readable to use `UNNEST(..) WITH ORDINALITY AS a(val, ord)` and then use explicit `a.val, a.ordinality` in SELECT clause rather this implicit and bit magical `ordinality` column coming out of nowhere?